hopsworksai/resource_cluster.go

```diff
@@ -386,6 +385,25 @@ func clusterSchema() map[string]*schema.Schema {
 			Default:      api.Ubuntu,
 			ValidateFunc: validation.StringInSlice([]string{api.Ubuntu.String(), api.CentOS.String()}, false),
 		},
+		"upgrade_in_progress": {
+			Description: "Information about ongoing cluster upgrade if any.",
+			Type:        schema.TypeList,
+			Computed:    true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"from_version": {
+						Description: "The source version from which the cluster is upgrading.",
```

Comment:
I think you can remove source and upgrade version. The version from/to which the cluster is upgrading is clear enough